loadtest/store/memstore/store.go

```diff
@@ -1356,6 +1359,53 @@ func (s *MemStore) DeleteChannelBookmark(bookmarkId string) error {
 	}
 
 	delete(s.channelBookmarks, bookmarkId)
+	return nil
+}
+
+func (s *MemStore) SetScheduledPost(teamId string, scheduledPost *model.ScheduledPost) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if scheduledPost == nil {
+		return errors.New("memstore: scheduled post should not be nil")
+	}
 
+	if s.scheduledPosts == nil {
+		s.scheduledPosts = map[string]map[string][]*model.ScheduledPost{}
+	}
+
+	if s.scheduledPosts[teamId] == nil {
+		s.scheduledPosts[teamId] = map[string][]*model.ScheduledPost{}
+	}
+
+	channelOrThreadId := scheduledPost.ChannelId
+	if scheduledPost.RootId != "" {
+		channelOrThreadId = scheduledPost.RootId
+	}
+
+	s.scheduledPosts[teamId][channelOrThreadId] = append(s.scheduledPosts[teamId][channelOrThreadId], scheduledPost)
 	return nil
 }
+
+func (s *MemStore) DeleteScheduledPost(scheduledPostID string) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	for teamId := range s.scheduledPosts {
+		if _, ok := s.scheduledPosts[teamId][scheduledPostID]; ok {
+			delete(s.scheduledPosts[teamId], scheduledPostID)
+			break
+		}
+	}
+}
+
+func (s *MemStore) UpdateScheduledPost(teamId string, scheduledPost *model.ScheduledPost) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if _, ok := s.scheduledPosts[teamId]; !ok {
+		s.scheduledPosts[teamId] = make(map[string][]*model.ScheduledPost)
+	}
+
+	s.scheduledPosts[teamId][scheduledPost.Id] = append(s.scheduledPosts[teamId][scheduledPost.Id], scheduledPost)
+}
```

Comment:
Why do we append to `scheduledPosts` instead of updating the post itself? Also, shouldn't we use the channel or thread ID instead of `scheduledPost.Id` in the inner map?

Reply:
Fixed.